DESIGN.md

```diff
@@ -0,0 +1,58 @@
+# Module Design
+
+## Classes
+
+There are 4 main public classes and other can be added if they need to be managed separately than the existing classes:
+
+* sensu - Resources shared by all other public classes such as agent and backend
+* sensu::agent - Sensu Agent
+* sensu::backend - Sensu Backend
+* sensu::plugins - Sensu plugins
+
+Subclasses to the above hold resources for that resources to keep the logic in the main public class shorter. So `sensu::backend::default_resources` is a private classes that adds functionality to `sensu::backend`. There are also some shared classes like `sensu::ssl` that are private and have resources shared by other public classes.
+
+If a parameter is used by both agent and backend, it belongs in `sensu` class. All other parameters should be added to the appropriate public class.
+
+## Types/Provides
```

Comment:
should document the helper functions under https://github.com/sensu/sensu-puppet/tree/master/lib/puppet_x/sensu

Reply:
Added section for these.